apps/opik-frontend/src/components/pages/PlaygroundPage/PlaygroundOutputs/PlaygroundOutputScores/PlaygroundOutputScores.tsx

```diff
@@ -0,0 +1,98 @@
+import React from "react";
+
+import {
+  HoverCard,
+  HoverCardContent,
+  HoverCardTrigger,
+} from "@/components/ui/hover-card";
+import { cn } from "@/lib/utils";
+import { TAG_VARIANTS_COLOR_MAP } from "@/components/ui/tag";
+import { generateTagVariant } from "@/lib/traces";
+import { FeedbackScoreValueByAuthorMap } from "@/types/traces";
+import MetricTag from "./MetricTag";
+
+const MAX_VISIBLE_METRICS = 3;
+
+// Score data from trace feedback
+export interface ScoreData {
+  value: number;
+  reason?: string;
+  lastUpdatedAt?: string;
+  lastUpdatedBy?: string;
+  valueByAuthor?: FeedbackScoreValueByAuthorMap;
+  category?: string;
+}
+
+interface PlaygroundOutputScoresProps {
+  metricNames: string[]; // Expected metric names from rules
+  metricScores: Record<string, ScoreData>; // Actual scores from trace (keyed by name)
+  stale?: boolean;
+  className?: string;
+}
+
+const PlaygroundOutputScores: React.FC<PlaygroundOutputScoresProps> = ({
+  metricNames,
+  metricScores,
+  stale = false,
+  className,
+}) => {
+  if (metricNames.length === 0) {
+    return null;
+  }
+
+  const metricColors = Object.fromEntries(
+    metricNames.map((name) => {
+      const variant = generateTagVariant(name);
+      return [name, TAG_VARIANTS_COLOR_MAP[variant ?? "gray"]];
+    }),
+  );
+
+  const visibleMetrics = metricNames.slice(0, MAX_VISIBLE_METRICS);
+  const hiddenMetrics = metricNames.slice(MAX_VISIBLE_METRICS);
+  const remainingCount = hiddenMetrics.length;
+
```

Comment:
No need to fix it now, but just for the future: 
It would be better to create them as part of `useMemo`.